sheer/indexer.py

```diff
@@ -134,44 +206,11 @@ def index_location(args, config):
         selected_processors = [p for p in processors if p.name in args.processors]
 
     for processor in selected_processors:
-        # If the mapping already exists, and we were called with the reindex
-        # flag, remove the mapping.
-        mapping = es.indices.get_mapping(index=index_name, doc_type=processor.name)
-        if mapping and args.reindex:
-            print "removing existing mapping for %s (%s)" % (processor.name, processor.processor_name)
-            es.indices.delete_mapping(index=index_name, doc_type=processor.name)
-            mapping = {}
-
-        # Then create the mapping if it does not exist
-        if not mapping:
-            print "creating mapping for %s (%s)" % (processor.name, processor.processor_name)
-            es.indices.put_mapping(index=index_name,
-                                doc_type=processor.name,
-                                body={processor.name: processor.mapping(default_mapping)})
-
-        i = -1
-        for i, document in enumerate(processor.documents()):
-            # Create the document. If it already exists in Elasticsearch an
-            # exception will be raised.
-            try:
-                es.create(index=index_name,
-                        doc_type=processor.name,
-                        id=document['_id'],
-                        body=document)
-            except TransportError, e:
-                # Elasticsearch status code 409 is DocumentAlreadyExistsException
-                # Anything else and we want to bail here.
-                if e.status_code != 409:
-                    raise e
-
-                # If the document couldn't be created because it already exists,
-                # update it instead.
-                es.update(index=index_name,
-                        doc_type=processor.name,
-                        id=document['_id'],
-                        body={'doc': document})
-
-            sys.stdout.write("indexed %s %s \r" % (i + 1, processor.name))
-            sys.stdout.flush()
-
-        sys.stdout.write("indexed %s %s \n" % (i + 1, processor.name))
+        # This could raise a value error if there's a problem reading the
+        # processor's source json. If that happens, continue on to the next
+        # content processor.
+        try:
+            index_processor(es, index_name, default_mapping, processor, reindex=args.reindex)
+        except ValueError:
```

Comment:
So this is caught again in `index_processor`, where it does print a message to stderr.